Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -839,8 +899,11 @@ ExceptionOr<void> MediaSource::removeSourceBuffer(SourceBuffer& buffer)
                 // 5.3.3 Remove the AudioTrack object from the HTMLMediaElement audioTracks list.
                 // 5.3.4 Queue a task to fire a trusted event named removetrack, that does not bubble and is not
                 // cancelable, and that uses the TrackEvent interface, at the HTMLMediaElement audioTracks list.
-                if (m_mediaElement)
-                    m_mediaElement->removeAudioTrack(track);
+                ensureOnMainThread([weakMediaElement = m_mediaElement, track = Ref { track }]() mutable {
+                    // FIXME: Need to send a mirror when we are in a worker.
+                    if (weakMediaElement)
```

Comment:
I would tend to RefPtr all the weakMediaElements here and below.

Reply:
they are made in RefPtr except where the operation is a one-liner

Reply:
did them all